[게시판 회원 기능] 슬로스 미션 제출 합니다.

soojinroh opened the pull request.

os94 committed:
[ADD] Add README

os94 committed:
[Fix] Code Formatting

os94 committed:
[Fix] Pass ArticleControllerTest, HelloWorldControllerTest

os94 committed:
[ADD] Move to writing page

os94 committed:
[ADD] Create article logic

os94 committed:
[ADD] Show article lists

os94 committed:
[ADD] Show selected article

os94 committed:
[ADD] Springboot Devtools (auto reload)

os94 committed:
[ADD] Remove duplication body code of index, article, article-edit HTML

- using Thymeleaf template layout

os94 committed:
[Fix] create_article Test

os94 committed:
[ADD] Remove duplicate head code of index, article, article-edit HTML

os94 committed:
[ADD] Modify article function

os94 committed:
[ADD] Remove article function

os94 committed:
[REF] Fix ArticleControllerTest

os94 committed:
[ADD] Redirect to article

os94 committed:
[REF] Add this keyword

os94 committed:
[REF] Extract magic number as constant variable

os94 committed:
[REF] Add matchId method in Article

os94 committed:
[ADD] ArticleRepository, ArticleController Test

os94 committed:
[REF] Improve ArticleRepository in ArticleController

- Modify field injection to constructor injection

os94 committed:
[Fix] add method in ArticleRepository

os94 committed:
[Fix] Remove this keyword

os94 committed:
[REF] Update 로직 수정, initial articleId 1로 수정

os94 committed:
[ADD] ArticleNotFoundException 추가

os94 committed:
[REF] ArticleControllerTests 중복제거

os94 committed:
Add : JPA 강의 적용

os94 committed:
Revert "Add : JPA 강의 적용"

This reverts commit f957b12deb23c177dd772a5430293452c62eaffb.

os94 committed:
Add : ArticleDto 생성

os94 committed:
Add : JPA-MySQL 설정 세팅

os94 committed:
Add : ArticleRepository 인터페이스 생성

os94 committed:
Refactor : Article class의 entity 속성 추가

os94 committed:
add : MySQL의 server time zone, UTF-8 encoding 설정 추가

os94 committed:
Refactor : 프론트 파일 교체

os94 committed:
Refactor : ArticleRepository를  CrudRepository를 상속하는 인터페이스로 변경

os94 committed:
Refactor : ArticleDto 삭제 및 ArticleController 수정

os94 committed:
Add : JPA Log 확인 설정 추가

os94 committed:
Refactor : ArticleControllerTests 수정

- 각 테스트마다 Article이 삽입 및 삭제되도록 수정

os94 committed:
Refactor : MainController 및 RequestMapping 분리

os94 committed:
Docs : README.md 수정

os94 committed:
Add : Login 페이지 이동 기능 구현

os94 committed:
Add : Signup 페이지 이동 기능 구현

os94 committed:
Add : User 클래스 생성

os94 committed:
Add : UserRequestDto 클래스 생성

os94 committed:
Add : UserRepository 인터페이스 생성

os94 committed:
Add : 회원가입 기능 구현

os94 committed:
Add : UserRequestDto Validation 패턴 추가

os94 committed:
Add : 회원가입 실패시 에러 메시지를 웹에 출력

os94 committed:
Add : UserControllerTests 회원가입 검증 테스트 추가

os94 committed:
Refactor : UserRequestDto 에러 메시지 상수화

os94 committed:
Add : signup.html 프론트 입력 검증 추가

os94 committed:
Add : signup.html 비밀번호 확인 기능 추가

os94 committed:
Add : 회원가입시 이메일 중복 확인 기능 추가

os94 committed:
Refactor : 이메일 중복 확인 기능 수정

os94 committed:
Add : 회원조회 기능 및 테스트 추가

os94 committed:
Add : 로그인, 로그아웃 기능 구현

os94 committed:
Refactor : 로그인 한 유저가 로그인시 메인화면으로 이동

os94 committed:
Refactor : Code Formatting

os94 committed:
Add, Refactor : 회원 탈퇴 기능 추가, 일부 메서드의 클래스 위치 변경

os94 committed:
Add : Logger 실습

os94 committed:
Refactor : HTML 중복 제거

os94 committed:
Add : user-list.html 중복 코드 제거

os94 committed:
Add : UserDetail 추가

os94 committed:
Add : mypage 이동 및 유저 정보 출력

os94 committed:
Add : mypage-edit 이동 및 유저 정보 출력

os94 committed:
Refactor : UserDetail 삭제

os94 committed:
Add : 회원정보 수정 기능 추가

os94 committed:
Add : 유저목록 탭 추가, delete버튼 UI 수정

os94 committed:
Add : 로그인여부에 따라 메인페이지 문구 출력

os94 committed:
Refactor : Controller 메서드 이전, 메서드 이름 수정

os94 committed:
Refactor : dto 패키지 분리

os94 committed:
Refactor : Controller에서 Service 분리

os94 committed:
Refactor : 사용하지않는 필드 삭제

os94 committed:
Refactor : 일부 테스트 메서드 이동 및 이름 수정

indiveloper committed:
revert

indiveloper committed:
Refactor : 패키지 및 파일 구조 변경

indiveloper committed:
Add : LoginControllerTests 구현

1. 로그인 성공
2. 로그인 실패(이메일 입력 오류)
3. 로그인 실패(패스워드 입력 오류)
4. 로그아웃 성공

indiveloper committed:
Refactor : package 구조 변경

indiveloper committed:
Add : ArticleServiceTests 구현

indiveloper committed:
Add : UserServiceTests 구현

indiveloper committed:
Add : LoginServiceTests 구현

indiveloper committed:
Refactor : annotation 수정

woowahan-pjs reviewed (changes requested):
오프라인 코드 리뷰로 대신하였습니다. 👍

indiveloper committed:
Refactor : User Entity 수정

1. setter 제거
2. 기본 생성자의 접근제한자 private으로 변경
3. equals를 id만 비교하는 것으로 변경
4. column annotaion에서 name 제거
5. table annotation에서 name 제거

indiveloper committed:
Refactor : Article Entity 수정

1. setter 제거
2. 기본 생성자의 접근제한자 private으로 변경
3. equals를 id만 비교하는 것으로 변경

indiveloper committed:
Refactor : LoginRepository 삭제

indiveloper committed:
Refactor : 불필요한 Annotation 삭제

indiveloper committed:
Refactor : 메서드 변경

userRepository.findUserByEmail을 userRepository.existsByEmail로 변경

indiveloper committed:
Refactor : package 위치 변경

indiveloper committed:
Refactor : Controller에서 하던 Transaction관리를 service로 변경

indiveloper committed:
Refactor : ArticleUpdateDto 생성

indiveloper committed:
Refactor : Id type 변경

long에서 Long으로 변경

indiveloper committed:
Refactor : 사용하지 않는 method 삭제

indiveloper committed:
Refactor : ERROR MESSAGE 별도의 클래스로 분리

indiveloper committed:
Refactor : package 위치 변경

indiveloper committed:
Refactor : Convention 수정

1. @Autowired 작성하는 것으로 통일
2. ==를 equals()로 변경

indiveloper committed:
Add : Error message를 모아놓은 Messages 클래스 구현

indiveloper committed:
Refactor : html 파일 수정

indiveloper committed:
Refactor : 사용하지 않는 test 삭제

indiveloper committed:
Refactor : 주석 제거

indiveloper committed:
Test : test 코드 수정

woowahan-pjs reviewed (approved):
스프링의 세계에 온 것을 환영합니다. 👍 
이만 Merge 하겠습니다.
